### DISCUSSION TITLE
Make it possible to add autoimported imports to the *first cell*

### DISCUSSION THREAD

**Discussion by umarbutler** [microsoft/pylance-release#7120] [Enhancement]
Say I'm in a notebook and I'm focused on cell 3. I type in `copy.d`, `copy.deepcopy()` appears as a suggestion, I then hit `tab`, and `import copy` automatically appears at the top of cell 3. I want to be able to have that import added to cell 1 instead.

The reason is that it is very annoying to have to copy my imports every single time they autoadd so much so that I don't even bother doing autoimports in Jupyter notebooks because it's easier to scroll up to cell 1 and add it.

Alternatively, could we have a way to move highlighted code up to cell 1? That way we can run `import copy` in cell 3 and then transport it to cell 1 after running it.

In terms of why I like my imports in cell 1 and not cell 3? Well, its the same reason people like to import at the top of their file instead of importing it as needed: not doing that is messy, confusing, and prevents you from deciding to go up to cell 2 and reference `copy`.

**Comment by debonte** [comment]
Moving this issue to discussions as an enhancement request for comments and upvotes.
